src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,77 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    public static final String SUPPLY = "supply,";
+    public static final String BUY = "buy,";
+    public static final String RESULT = "result,";
+    public static final String DELIMETER = ",";
+    private StringBuilder stringBuilder = new StringBuilder();
+    private int buy = 0;
+    private int supply = 0;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        readData(fromFileName);
+        extracted(supply,buy);
+        writeData(buy, supply, toFileName, stringBuilder.toString());
+    }
+
+    private void extracted(int supply, int buy) {
+        stringBuilder.append(SUPPLY)
+                    .append(supply)
+                    .append(System.lineSeparator())
+                    .append(BUY)
+                    .append(buy)
+                    .append(System.lineSeparator())
+                    .append(RESULT)
+                    .append(supply - buy)
+                    .append(System.lineSeparator());
```

Comment:
Actually here you can simply use string concatenations instead of stringBuilder as there is no loop and the compiler will probably optimize code execution